src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java

```diff
@@ -326,6 +330,50 @@ public boolean isTableCompatible(final String tableName) {
     return compatible;
   }
 
+  @Override
+  public void appendMetaColIfNotExist(final String tableName) {
+    checkConnection();
+    InternalUtils.assertNotEmpty("tableName", tableName);
+    String query = "desc table identifier(?)";
+    PreparedStatement stmt = null;
+    ResultSet result = null;
+    boolean hasMeta = false;
+    boolean isVariant = false;
+    try {
+      stmt = conn.prepareStatement(query);
+      stmt.setString(1, tableName);
+      result = stmt.executeQuery();
+      while (result.next()) {
+        // The result schema is row idx | column name | data type | kind | null? | ...
+        if (result.getString(1).equals(METADATA_COLUMN)) {
+          hasMeta = true;
+          if (result.getString(2).equals("VARIANT")) {
+            isVariant = true;
+          }
+          break;
+        }
+      }
+    } catch (SQLException e) {
+      logError("table {} doesn't exist", tableName);
+      throw SnowflakeErrors.ERROR_2014.getException("table name: " + tableName);
+    }
+    try {
+      if (!isVariant) {
+        String metaQuery;
+        if (!hasMeta) {
+          metaQuery = "alter table identifier(?) add RECORD_METADATA VARIANT";
```

Comment:
Do we need to make any changes in the docs when we publish about schema mapping, about some specific privilege/permissions?

Reply:
The assumption is that the user themself needs to create the table with the correct schema. They will need the privilege to alter the table if the metadata column is added by us. In that case they need one of 'INSERT', 'ALL' or 'OWNERSHIP'.